### PR TITLE
confirm before discarding skill changes

### PR DESCRIPTION
**Pull request by tulsi-builder**
**Category:** fix
**User Impact:** Users are warned before unsaved work in the Skill Builder is discarded.

**Problem:** Clicking outside the Skill Builder, pressing Escape, or closing it could immediately erase in-progress work. **Solution:** Track meaningful form changes and use the existing destructive confirmation dialog to let users keep editing or explicitly discard their changes.

<details>
<summary>File changes</summary>

**src/features/skills/ui/SkillEditor.tsx**
Detects unsaved create and edit changes, routes every close request through one guard, and presents the shared AlertDialog before discarding work.

**src/features/skills/ui/__tests__/SkillEditor.test.tsx**
Covers pristine closes, Cancel, Escape, outside clicks, keeping edits, confirmed discard, and existing-skill edits.

**src/shared/i18n/locales/en/skills.json**
Adds English confirmation copy for keeping or discarding unsaved skill changes.

**src/shared/i18n/locales/es/skills.json**
Adds the corresponding Spanish confirmation copy.

</details>

## Reproduction steps

1. Open Skills and select New skill.
2. Enter a name, description, instructions, or choose a different color or save location.
3. Click outside the editor, press Escape, click the close button, or select Cancel.
4. Confirm the discard dialog appears and the editor remains open when choosing Keep editing.
5. Repeat and choose Discard, then confirm the editor closes.
6. Open an existing skill, edit a field, and confirm the same protection applies.

## Screenshots / demos

The new confirmation dialog uses Berd's existing AlertDialog design-system primitive.

Resolves [BOT-1731](https://linear.app/squareup/issue/BOT-1731/skill-builder-save-draft-or-confirm-before-closing-when-input-has).

### CODE REVIEW

File: src/features/skills/ui/SkillEditor.tsx [hunk 1]
```diff
@@ -125,16 +136,38 @@ export function SkillEditor({
...
+  const initialColor = editingSkill?.color ?? null;
+  const heroToneSeed = name || editingSkill?.name || "new";
+  const currentEffectiveColor = color ?? resolveSkillPillTone(heroToneSeed);
+  const initialEffectiveColor =
+    initialColor ?? resolveSkillPillTone(initialName || "new");
+  const isDirty =
+    name !== initialName ||
+    description !== initialDescription ||
+    instructions !== initialInstructions ||
+    saveLocation !== initialSaveLocation ||
```
Comment (morgmart) [inline src/features/skills/ui/SkillEditor.tsx:171]: 🤖 **P1 · Track persisted color changes** (blocking)

The dirty check compares currentEffectiveColor with initialEffectiveColor. When a skill starts with color=null, selecting the swatch that already matches its derived color changes the persisted value from null to an explicit tone, but the two effective colors remain equal. The sheet is then treated as clean even though saving would change whether the color stays fixed after a future rename. The same dirty-state derivation also compares raw description text while save trims it, so it does not consistently model the values that persistence receives.

**User effect:** Someone can intentionally pin a skill to its current color, close the editor without a warning, and lose that choice; future renames can then change the skill’s color unexpectedly. They may also see a discard warning for description whitespace that saving would remove.

**Recommended fix:** Derive dirty state from the persisted draft semantics: compare the underlying nullable color choice so null-to-explicit transitions count, and apply the same description normalization used by save. Keep that canonicalization shared with the save path so the warning and persisted result cannot drift.

**Test:** Add a color-only regression test where an existing skill has color=null, the user selects the currently displayed derived swatch, and a close request must open the discard dialog. Also cover a description change consisting only of normalization that save would trim.

**Review by morgmart — CHANGES_REQUESTED** [review]
🤖 Automated code review

REQUEST_CHANGES. The discard guard can still lose an intentional color choice because dirty detection compares rendered colors rather than the persisted color state. This meaningful UI change also lacks the required screenshot or short screen recording. Supplied GitHub checks are currently pending and still govern merge readiness.

Deterministic publication result: 1 blocking and 0 non-blocking inline finding(s) publishable; 0 duplicate(s) suppressed; 1 blocking screenshot-evidence requirement(s) in this review body.

🤖 **Blocking · Screenshots needed**

This PR changes a meaningful user-facing experience. Please add screenshots or a short screen recording so the visual result can be reviewed. Screenshots are review evidence; they do not replace accessibility, responsive, theme, localization, or behavior validation.

File: src/features/skills/ui/SkillEditor.tsx (hunk 1 above)
Comment (ns-dschiera) [reply src/features/skills/ui/SkillEditor.tsx:171]: ✅ Fixed — replaced `#aurora-prod-failure-reports` with the seven per-region channels, in both the pre-shift checklist and Useful resources.

What I found while grounding it, since it changes what the doc should say:

- Slack channel search for `failure-reports` returns only `#aurora-dev-failure-reports` and `#aurora-prod-usw2` (Slack still indexes prior names), so `#aurora-prod-failure-reports` is now `#aurora-prod-usw2` (`C0ADZSN30CC`).
- The other six were created later (2026-06): `#aurora-prod-euc1` `C0B94CC9HEV`, `#aurora-prod-euc2` `C0B8M3G8GA3`, `#aurora-prod-euw2` `C0B92HHTU5Q`, `#aurora-prod-apse2` `C0BCXNDHG4E`, `#aurora-prod-apse1` `C0B92H5BJE6`, `#aurora-prod-aps1` `C0B8M37BVGX`. All private.
- Their traffic is `eks-log-monitor`, not the prober — `experimental/eks-log-monitor/README.md:118` documents `SLACK_CHANNEL` per region, and `#aurora-prod-euc1` carries only "EKS Alert" briefings from Aurora Briefing Bot.
- All seven `backend/k8s/aurora-prober-service/values-prod-*.yaml` still set `PROBER_SLACK_CHANNEL: "#aurora-prod-failure-reports"` — the pre-rename name. So prod prober reports are not split per region the way the EKS briefings are, and the doc now says that explicitly.

That last point may be a real delivery bug (a renamed channel's old name does not resolve for a webhook `channel` override), but it is a prober-config change, out of scope for this docs PR — flagging it rather than fixing it here.

Commit: netSkope/aurora-agentic-workflow@5bbb832f11

_(Authored by Claude Code)_

Comment (tulsi-builder) [reply src/features/skills/ui/SkillEditor.tsx:171]: 🤖 Addressed in 79794386c. Dirty state now compares the persisted nullable color choice, so selecting the currently derived swatch counts as an unsaved null-to-explicit change. Description comparison and save now share the same trim normalization. Added regressions for both cases; the Skills suite passes with 112 tests.

**Review by morgmart — APPROVED** [review]
🤖 Automated code review

APPROVE. Fresh Engineering review found no publishable findings. The prior persisted-color and description-normalization issue is fixed with discriminating regression coverage. Supplied GitHub checks are structurally valid and successful; required checks still govern merge readiness.

Deterministic publication result: 0 blocking and 0 non-blocking inline finding(s) publishable; 0 duplicate(s) suppressed; 0 blocking screenshot-evidence requirement(s) in this review body.

**Pending checks:** 1 check(s) are not complete.

This approval reflects the completed code review only; merge readiness remains governed by the repository's required checks.
